packages/headless/src/features/query-suggest/query-suggest-slice.test.ts

```diff
@@ -186,15 +188,19 @@ describe('querySuggest slice', () => {
         );
         expect(nextState[id]?.partialQueries).toEqual([q]);
       });
-    });
-  });
 
-  it('#executeSearch.fulfilled causes the partialQueries array to be reset', () => {
-    const query = 'query';
-    const initialState = addToDefaultState({partialQueries: [query]});
-    const search = buildMockSearch();
-    const action = executeSearch.fulfilled(search, '', logSearchboxSubmit());
-    const finalState = querySuggestReducer(initialState, action);
-    expect(finalState[id]?.partialQueries).toEqual([]);
+      it('should encode `;` characters in the list of partialQueries', () => {
```

Comment:
Since the value is ultimately sent as a `;` separated string, this needs to be escaped.

We could improve this in `coveo.analytics.js`

Reply:
Indeed, I think any escaping should be handled by `coveo.analytics.js`. However, I'm not clear on why it needs to be escaped? Is the payload being sent in the url using a GET request?

Reply:
No, it's because `['he', 'hel', 'hello']` as partialQueries will ultimately be sent as `he;hel;hello` to UA/ML, inside coveo.analytics.js.

If we don't escape `[';', ';he', ';hello']` to `['%3B', '%3Bhe', '%3Bhello']` ML will get the wrong input (`;;;he;;hello`)